com.chartboost.helium/Android~/helium-android-bridge/src/main/kotlin/com/chartboost/heliumsdk/unity/HeliumUnityAdWrapper.kt

```diff
@@ -0,0 +1,236 @@
+        if (_ad == null) throw RuntimeException("$TAG cannot interact with Helium ad as ad was not created")
+        return _ad
+    }
+
+    init {
+        if (isBanner) _asBanner = _ad as HeliumBannerAd?
+        if (isFullScreen) _asFullScreen = _ad as HeliumFullscreenAd?
+    }
+
+    private fun asBanner(): HeliumBannerAd? {
+        return _asBanner
+    }
+
+    private fun asFullScreen(): HeliumFullscreenAd? {
+        return _asFullScreen
+    }
+
+    fun load() {
+        HeliumUnityBridge.runTaskOnUiThread {
+            ad().load()
+            startedLoad = true
+        }
+    }
+
+    fun load(screenLocation: Int) {
+        HeliumUnityBridge.runTaskOnUiThread {
+            createBannerLayout(screenLocation)
+            load()
+        }
+    }
+
+    fun show() {
+        HeliumUnityBridge.runTaskOnUiThread { if (isFullScreen) asFullScreen()!!.show() }
+    }
+
+    fun setKeyword(keyword: String?, value: String?): Boolean {
+        val keywords = ad().keywords // not null by design
+        return keywords.set(keyword!!, value!!)
+    }
+
+    fun removeKeyword(keyword: String?): String? {
+        val keywords = ad().keywords // not null by design
+        return keywords.remove(keyword!!)
+    }
+
+    fun setCustomData(customData: String?) {
+        val ad = ad()
+        if (ad is HeliumRewardedAd) {
+            ad.customData = customData
+        } else {
+            throw RuntimeException("$TAG custom data can only be set on a rewarded ad")
+        }
+    }
+
+    fun setBannerVisibility(isVisible: Boolean) {
+        HeliumUnityBridge.runTaskOnUiThread {
+            if (isBanner && mBannerLayout != null) {
+                val visibility = if (isVisible) View.VISIBLE else View.INVISIBLE
+                mBannerLayout!!.visibility = visibility
+                asBanner()!!.visibility = visibility
+            }
+        }
+    }
+
+    fun clearLoaded(): Boolean {
+        if (isFullScreen) return asFullScreen()!!.clearLoaded()
+        if (isBanner) {
+            HeliumUnityBridge.runTaskOnUiThread { asBanner()!!.clearAd() }
+            return true
+        }
+        return false
+    }
+
+    fun readyToShow(): Boolean {
+        return if (!startedLoad) false else try {
+            if (isFullScreen) asFullScreen()!!.readyToShow() else if (isBanner) {
+                Log.w(TAG, "This should never be called, banners do not have readyToShow")
+                false
+            } else false
+        } catch (ex: Exception) {
+            Log.w(TAG, "Helium encountered an error calling Ad.readyToShow() - " + ex.message)
+            false
+        }
+    }
+
+    fun destroy() {
+        HeliumUnityBridge.runTaskOnUiThread {
+            destroyBannerLayout()
+            ad().destroy()
+        }
+    }
+
+    private fun createBannerLayout(screenLocation: Int) {
+        if (!isBanner) return
+        if (mBannerLayout != null && mBannerLayout!!.childCount >= 0) mBannerLayout!!.removeAllViews()
```

Comment:
mBannerLayout could have changed between your first and second call.
Either assign a local variable
```
val localBannerLayout = mBannerLayout ?: RelativeLayout(_activity).apply {
    setBackgroundColor(Color.TRANSPARENT)
// ...
mBannerLayout = localBannerLayout
```
or use a let. but here, since you reassign it, you might want to keep it around.